transports/dns/src/config/async_std.rs

```diff
@@ -0,0 +1,32 @@
+use std::{io, sync::Arc};
+
+use async_std_resolver::AsyncStdResolver;
+use parking_lot::Mutex;
+use trust_dns_resolver::{
+    config::{ResolverConfig, ResolverOpts},
+    system_conf,
+};
+
+/// A `Transport` wrapper for performing DNS lookups when dialing `Multiaddr`esses
+/// using `async-std` for all async I/O.
+pub type Config<T> = crate::Config<T, AsyncStdResolver>;
```

Comment:
Let's name these `Transport` to be consistent with TCP, QUIC, etc

```suggestion
pub type Transport<T> = crate::Config<T, AsyncStdResolver>;
```